lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -324,6 +326,36 @@ def post(self, request):
             uname = request.POST.get('student_uname', '').strip()
             self.msg = u'<h4>{0}</h4><p>{1}</p><hr />{2}'.format(
                 _('Delete User Results'), self.delete_user(uname), self.msg)
+        elif action == 'rename_user':
+            old_uname = request.POST.get('old_uname', '').strip()
+            new_uname = request.POST.get('new_uname', '').strip()
+            if len(old_uname) == 0 or len(new_uname) == 0:
+                message = 'Usernames cannot be blank'
+            else:
+                try:
+                    rename_user(old_uname, new_uname)
+                except (User.DoesNotExist):
+                    message = 'User \'{old_username}\' does not exist'.format(
+                        old_username=old_uname
+                    )
+                except (IntegrityError):
+                    message = 'User \'{new_username}\' already exists'.format(
+                        new_username=new_uname
+                    )
+                except (PyMongoError):
+                    message = 'Failed to modify username for user \'{old_username}\''.format(
+                        old_username=old_uname
+                    )
+                else:
+                    message = "Changed username of user \'{old_username}\' to \'{new_username}\'".format(
+                        old_username=old_uname,
+                        new_username=new_uname,
+                    )
```

Comment:
I think we should move this block (everything up to this line) into a separate helper function `_rename_user`, similar to how we're already doing it above with `self.delete_user` and `self.create_user`.